scene/main/node.cpp

```diff
@@ -1289,6 +1289,31 @@ Node *Node::_get_child_by_name(const StringName &p_name) const {
 	}
 }
 
+int Node::get_descendant_count(bool p_include_internal) const {
+	int count = 0;
+	TypedArray<Node> children = get_children(p_include_internal);
```

Comment:
I'm not sure using the "get_children" function is the best way to achieve this, none of the other functions use this and instead access the children directly, see the function itself for that, not sure what the performance difference would be but that would be my suggestion